### PR TITLE
move assets options from templating to assets section and add base_path documentation

### PR DESCRIPTION
**Pull request by snoek09**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.7 |
| Fixed tickets | #6146 |

### CODE REVIEW

**Comment by snoek09** [comment]
@xabbuh please let me know if this is OK.

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -850,14 +851,13 @@ setting the value to ``null``:
             ),
         ));
 
-templating
-~~~~~~~~~~
+assets
+~~~~~~
 
 .. _reference-framework-assets-version:
-.. _ref-framework-assets-version:
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: I would not remove this anchor (there may be external links to it).

**Comment by michaelperrin** [comment]
This should definitely be merged as the current situation is misleading.

Is there a PR on the Symfony project itself? The default `config.yml` file has the following code:

```
templating:
        engines: ['twig']
        #assets_version: SomeVersionScheme
```

But uncommenting `assets_version` just generates an error.

**Comment by xabbuh** [comment]
There is a pull request: symfony/symfony-standard#891

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -888,7 +889,10 @@ Now, activate the ``assets_version`` option:
         # app/config/config.yml
         framework:
             # ...
-            templating: { engines: ['twig'], assets_version: v2 }
+            assets:
+                version: 'v2'
+            templating:
+                engines: ['twig']
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: We can omit the `templating` node entirely here (same for the other formats).

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -911,40 +916,42 @@ Now, activate the ``assets_version`` option:
...
 option.
 
 .. tip::
 
     As with all settings, you can use a parameter as value for the
-    ``assets_version``. This makes it easier to increment the cache on each
+    ``version``. This makes it easier to increment the cache on each
     deployment.
 
-.. _reference-templating-version-format:
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: The old label should be kept.

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -1277,7 +1287,7 @@ Each package can configure the following options:
 
 * :ref:`base_urls <reference-templating-base-urls>`
 * :ref:`version <reference-framework-assets-version>`
-* :ref:`version_format <reference-templating-version-format>`
+* :ref:`version_format <reference-assets-version-format>`
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: The complete block about packages should be moved to the `assets` node too.

Comment (xabbuh) [inline reference/configuration/framework.rst]: and the same applies to the base url options

**Comment by snoek09** [comment]
I removed the examples with protocol-relative URLs. Do we want to keep this and rephrase it?

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -69,19 +69,18 @@ Configuration
     * `gc_probability`_
     * `gc_maxlifetime`_
     * `save_path`_
+* `assets`_
+    * `version`_
+    * `version_format`_
+    * `base_urls`_
+    * `packages`_
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: We should sort the options alphabetically but we can do that when merging.

**Comment by xabbuh** [comment]
Looks really great to me. Actually the only thing that is missing no is the description of the `base_path` option.

File: reference/configuration/framework.rst (hunk 5 above)
Comment (snoek09) [inline reference/configuration/framework.rst]: I kept it in the same order as here: http://symfony.com/blog/new-in-symfony-2-7-the-new-asset-component
Also the alphabetic order is not consistent in the framework configuration reference.

**Comment by snoek09** [comment]
Thank you @xabbuh. Do we want to add the description for `base_path` in this PR or separately?

**Comment by xabbuh** [comment]
@snoek09 If you happen to find some time to add it, that would be great. Otherwise, I think we can merge this as is and it afterwards in a separate PR.

**Comment by snoek09** [comment]
@xabbuh I've added the `base_path` option. Maybe this option needs some more text explaining it.

**Comment by weaverryan** [comment]
Wow, *great* work on this - very thorough. Thanks Henry!

**Pull request merged**